README.md

```diff
@@ -1 +1,588 @@
+
+    /** 
+     * Since we now know that microtasks run after JavaScript has finished
+     * executing, we can use this knowledge to our advantage, and only set 
+     * the `updateRequested` flag back to `false` again once all the tasks 
+     * have run, essentially delaying, or _batching_ the update!
+     */
+    Promise.resolve().then(() => {
+      updateRequested = false;
+      update();
+    });
+  }
+}
+
+/* This is our `update` function that we only want to be called once */
+function update() {
+  console.log('updating!');
+}
+
+/* We call scheduleUpdate in quick succession */
+scheduleUpdate();
+scheduleUpdate();
+scheduleUpdate();
+
+// "updating!" only logged once
+```
+
+What we do here is essentially _gatekeep_ potentially multiple incoming calls for `scheduleUpdate` with the `if` statement, and cleverly use our newfound knowledge of microtasks to our advantage to process any other incoming calls _first_, and only once JavaScript has finished executing, and when its microtask processing time, our `update` function is called.
+
+As a fun tidbit of knowledge, we can also write the `scheduleUpdate` function like so:
+
+```js
+function scheduleUpdate() {
+  if(!updateRequested) {
+    updateRequested = true;
+    updateRequested = await false;
+    update();
+  }
+}
+```
+
+Since, from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await):
+> If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise.
+
+In fact, [Justin Fagnani](https://twitter.com/justinfagnani), one of the authors of LitElement, once told me on twitter that the pre-LitElement version did exactly this:
+
+![twitter screenshot](./img/justin.png)
+
+### Deferring promises
+
+Alright, we're almost getting to the good stuff now. But before we go there, there's one final concept we need to explore, and thats deferring resolving of Promises, which will provide us with a handy utility later on in this blog post.
+
+Take a look at the following code:
+
+```js
+/* We declare a new variable */
+let resolveFn;
+
+/** 
+ * We declare a new promise, but we _dont_ resolve it! 
+ * Instead, we assign the `resolve` function to the
+ * `resolveFn` that we declared above
+ */
+const myPromise = new Promise((resolve) => {
+  resolveFn = resolve;
+});
+
+/** 
+ * `myPromise` now is just a pending promise, waiting to be resolved,
+ * and we can do that whenever we please with our `resolveFn`
+ */
+myPromise; // Promise {<pending>}
+
+/* For example, we could do a whole bunch of other work in between */
+
+/* And finally, whenever we're ready, we can call the `resolveFn` to resolve `myPromise` */
+resolveFn(); // Resolves `myPromise`
+
+/* Aaand `myPromise` is now fulfilled! */
+myPromise; // Promise {<fulfilled>: undefined} ! 🎉
+```
+
+This pattern is often used in cases (like unit tests) where you have to await a specific amount of time before continuing.
+
+```js
+function sleep(delay) {
```

Comment:
I'm not sure sleep is the best example, because you're passing resolve directly in the body of the promise, where as with `updateComplete` the fun starts when you reassign resolve to something out of scope